### ISSUE TITLE
Two-factor authentication not working

### ISSUE THREAD

**Issue by xvitaly** [Aseman-Land/Cutegram#236]
Trying to login but got an error `Error 400: password hash invalid`.

**Comment by Lesik** [comment]
I have the same issue.

**Comment by SG5** [comment]
I just have nothing after login code enter.
Cutegram 2.7.1 @ osx 10.11.6

**Comment by logs777** [comment]
I have the same issue on Windows 10 with client cutegram version 2.7.1.
If I'm using account without 2FA configured, everything works except secret chats (I can receive messages in secret chat, but my messages doesn't come to iphone official client).
If I'm using account with 2FA configured, I can't login - I don't see field for passing 2FA passphrase., only button "Resend Code"

**Comment by andersbateva** [comment]
Duplicate of #46

**Comment by AforAnonymous** [comment]
Not quite, #46 is for 2.7.1, this is for 3.0 alpha. Try again please @xvitaly, if you can get it to build. @flohack74 made some commits:

https://github.com/Aseman-Land/TelegramQML/pull/65
https://github.com/Aseman-Land/libqtelegram-aseman-edition/pull/58

that should fix this, see this comment: https://github.com/Aseman-Land/Cutegram/issues/46#issuecomment-335093408
